woocommerce/woocommerce-ios
# [SiOC - customer creation] Add feature flag for new customer section UI

Closes #12591 
Part of #12592 


## Why

As we started implementing the UI/UX changes to the order form customer section, we want to do it in a way so as not to break the existing flow in production. In this PR, a feature flag was created and used to show two different SwiftUI views based on the value.

## How

Since the redesigned UI/UX is almost completely different from the existing flow in the app, I decided to rename the existing `OrderCustomerSection` to `LegacyOrderCustomerSection` & move it to a separate folder, and use `OrderCustomerSection` for the new flow. The new flow is currently the same as before, except for the title of the Add Customer row and always separating the customer & note sections. In future PRs, changes to the new flow will be made incrementally.

## Testing instructions


- [x] @jaclync tests that the customer flow works as before when the feature flag is off
---
- Go to the orders tab
- Tap + to create an order --> `Add Customer` row should be shown (used to be `Add Customer Details`) in a different section from `Add Note`
- Feel free to test adding a customer
- Go back to the orders tab
- Tap on an order with a customer set
- Tap `Edit` --> the existing customer info should be shown

## Screenshots


before | after
-- | --
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/e1995c57-0ba7-4339-be7a-63a3ffa0b233" width="300" /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/ee65499f-cf3a-4f4f-9576-794ca5b85d48" width="300" />

---
